app/Console/Commands/AdsProcessTx.php

```diff
@@ -0,0 +1,198 @@
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * AdServer is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty
+ * of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with AdServer. If not, see <https://www.gnu.org/licenses/>
+ */
+
+namespace Adshares\Adserver\Console\Commands;
+
+use Adshares\Ads\AdsClient;
+use Adshares\Ads\Driver\CommandError;
+use Adshares\Ads\Entity\Transaction\SendManyTransaction;
+use Adshares\Ads\Entity\Transaction\SendManyTransactionWire;
+use Adshares\Ads\Entity\Transaction\SendOneTransaction;
+use Adshares\Ads\Exception\CommandException;
+use Adshares\Adserver\Facades\DB;
+use Adshares\Adserver\Models\AdsTxIn;
+use Adshares\Adserver\Models\User;
+use Adshares\Adserver\Models\UserLedger;
+use Illuminate\Console\Command;
+
+class AdsProcessTx extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'ads:process-tx';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Processes incoming txs';
+
+    /**
+     * Command ended without error
+     */
+    const EXIT_CODE_SUCCESS = 0;
+
+    /**
+     * Command ended prematurely because block ids could not be updated
+     */
+    const EXIT_CODE_CANNOT_GET_BLOCK_IDS = 1;
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        parent::__construct();
+    }
+
+    /**
+     * Execute the console command.
+     * @param AdsClient $adsClient
+     * @return int
+     */
+    public function handle(AdsClient $adsClient)
+    {
+        try {
+            $this->updateBlockIds($adsClient);
+        } catch (CommandException $exc) {
+            $code = $exc->getCode();
+            $message = $exc->getMessage();
+            $this->error("Cannot update blocks due to CommandException:\n"
+                . "Code:\n  ${code}\n"
+                . "Message:\n  ${message}\n");
+            return self::EXIT_CODE_CANNOT_GET_BLOCK_IDS;
+        }
+
+        $adServerAddress = config('app.adshares_address');
+        $txs = AdsTxIn::where('status', AdsTxIn::STATUS_NEW)->get();
+        /** @var $tx AdsTxIn */
+        foreach ($txs as $tx) {
+            $transaction = $adsClient->getTransaction($tx->txid)->getTxn();
+            $type = $transaction->getType();
+            switch ($type) {
+                case 'send_many':
+                    $isTxTargetValid = false;
+                    /** @var $sendManyTx SendManyTransaction */
+                    $sendManyTx = $transaction;
+//                    $senderAddr = $sendManyTx->getSenderAddress();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS125)